internal/pkg/service/buffer/api/service/mapper/task.go

```diff
@@ -0,0 +1,34 @@
+package mapper
+
+import (
+	"github.com/keboola/keboola-as-code/internal/pkg/service/buffer/api/gen/buffer"
+	"github.com/keboola/keboola-as-code/internal/pkg/service/buffer/store/model"
+)
+
+func (m Mapper) TaskPayload(model *model.Task) (r *buffer.Task) {
+	var finishedAtPtr *string
+	if model.FinishedAt != nil {
+		finishedAtStr := model.FinishedAt.String()
+		finishedAtPtr = &finishedAtStr
+	}
+	var durationMs int64 = 0
+	if model.Duration != nil {
+		durationMs = model.Duration.Milliseconds()
+	}
+	var errPtr *string
+	if model.Error != "" {
+		errPtr = &model.Error
+	}
+	return &buffer.Task{
+		ID:         model.TaskID,
+		ReceiverID: model.ReceiverID,
+		URL:        formatTaskURL(m.bufferAPIHost, model.TaskKey),
+		Type:       model.Type,
+		CreatedAt:  model.CreatedAt.String(),
+		FinishedAt: finishedAtPtr,
+		Duration:   durationMs,
+		IsFinished: model.IsFinished(),
+		Result:     model.Result,
+		Error:      errPtr,
+	}
+}
```

Comment:
Tu by bolo dobre doriesit este ostatne fields, ktore pri beziacom tasku nedavaju zmysel:

```suggestion
func (m Mapper) TaskPayload(model *model.Task) (r *buffer.Task) {
	var finishedAtPtr *string
	if model.FinishedAt != nil {
		v := model.FinishedAt.String()
		finishedAtPtr = &v
	}

	var resultPtr *string
	var errorPtr *string
	if model.Error == "" {
		resultPtr = &model.Result
	} else {
		errorPtr = &model.Error
	}

	var durationMsPtr *int64
	if model.Duration != nil {
		v := model.Duration.Milliseconds()
		durationMsPtr = &v
	}

	return &buffer.Task{
		ID:         model.TaskID,
		ReceiverID: model.ReceiverID,
		URL:        formatTaskURL(m.bufferAPIHost, model.TaskKey),
		Type:       model.Type,
		CreatedAt:  model.CreatedAt.String(),
		FinishedAt: finishedAtPtr,
		Duration:   durationMsPtr,
		IsFinished: model.IsFinished(),
		Result:     resultPtr,
		Error:      errorPtr,
	}
}
```

Reply:
ok 👍